src/NuGet.Configuration/Utility/RuntimeEnvironmentHelper.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.IO;
+
+namespace NuGet.Configuration
+{
+    internal static class RuntimeEnvironmentHelper
+    {
+        private static Lazy<bool> _isMono = new Lazy<bool>(() => Type.GetType("Mono.Runtime") != null);
+
+        // Checking the OS environment variable is most complete way to check this on dnxcore currently.
+        private static Lazy<bool> _isWindows = new Lazy<bool>(() => 
```

Comment:
Think you could get away with doing eagerly evaluating this

`private static readonly bool _isWindows = Environment.GetEnvironmentVariable("OS")?.Equals("WINDOWS_NT") ?? false`